subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[authoring_maintainable_build_scripts]]
+== Authoring Maintainable Build Scripts
+
+Gradle build scripts combine the qualities of declarative build logic, expressiveness as well as flexibility and rigidity as needed. As a build script author it is easy to fall into the trap of striking the wrong balance or applying poor coding habits. This chapter describes best practices for writing your build script in a meaningful, yet flexible and efficient way.
+
+[NOTE]
+====
+The third-party link:https://github.com/nebula-plugins/gradle-lint-plugin[Gradle lint plugin] helps with enforcing a desired code style in a build script if you are looking for appropriate link:https://en.wikipedia.org/wiki/Lint_(software)[linting] automation.
+====
+
+[[sec:declare_task_group_and_description]]
+=== Declare the group and description for tasks
+
+Even new users to a build should to be able to find crucial information quickly and effortlessly. In Gradle you can declare a api:org.gradle.api.Task#setGroup(java.lang.String)[group] and a api:org.gradle.api.Task#setDescription(java.lang.String)[description] for any task of the build. The <<sec:listing_tasks,tasks report>> uses the assigned values to organize and render the task for easy discoverability. Assigning a group and description is most helpful for lifecycle tasks.
+
+The example task `generateDocs` generates documentation for a project in the form of HTML pages. The task should be organized underneath the bucket `Documentation`. The description should express its intent.
+
+++++
+<sample id="taskGroupDescription" dir="userguide/bestPractices/taskGroupDescription" title="A task declaring the group and description">
+    <sourcefile file="build.gradle"/>
+</sample>
+++++
+
+The output of the tasks report reflects the assigned values.
+
+----
+$ gradle tasks
+
+> Task :tasks
+
+...
+
+Documentation tasks
+-------------------
+generateDocs - Generates the HTML documentation for this project.
+
+...
+----
+
+[[sec:avoid_use_of_imperative_logic]]
+=== Avoid the use of imperative logic
```

Comment:
I think "Avoid imperative logic in scripts" might be a bit clearer. "the use of" is probably not necessary.

Reply:
done